src/Utils/time.ts

```diff
@@ -1,46 +1,40 @@
-import { metaphysics } from "./metaphysics"
+import { timeQueryResponse } from "__generated__/timeQuery.graphql"
+import { graphql } from "react-relay"
+import { Environment, fetchQuery } from "relay-runtime"
 
 const getLocalTimestampInMilliSeconds = () => {
   return Date.now()
 }
 
-interface SystemTimeResults {
-  data: {
-    system: {
-      time: {
-        unix: number
+export async function getOffsetBetweenGravityClock(
+  relayEnvironment: Environment
+): Promise<number> {
+  const query = graphql`
+    query timeQuery {
+      system {
+        time {
+          unix
+        }
       }
     }
-  }
-}
+  `
 
-const fetchSystemTime = () => {
-  return metaphysics<SystemTimeResults>({
-    query: `
-      {
-        system {
-          time {
-            unix
-          }
-        }
-      }
-    `,
-  })
-}
+  const fetchSystemTime = () => {
+    return fetchQuery<{ data: timeQueryResponse }>(relayEnvironment, query, {})
```

Comment:
@damassi Realized an important detail here, this needs to specify a cache config that forces a refetch. Even if we don’t cache responses in Reaction yet, there will come a time and this would break.

Reply:
Will update 👍

Reply:
Thanks!

Reply:
Looking at this, there doesn't seem to be an option to configure cache (maybe because of its low-level-ness?) See https://relay.dev/docs/en/fetch-query.

Reply:
Can you look at the source instead? Perhaps the docs are wrong, otherwise the typings are.

Reply:
❗️ Always look at source: https://github.com/facebook/relay/blob/master/packages/relay-runtime/query/fetchQuery.js#L32 -- it's the last slot. I'll update the docs as well.

Reply:
I’m pretty sure the types In Emission has this, so perhaps it’s just a matter of updating them during the migration?

The docs need a fix indeed 👍

Reply:
https://github.com/facebook/relay/pull/2881

Looking at DT it's there, just need to update reaction.